api/src/services/graphql/schema-cache.ts

```diff
@@ -0,0 +1,16 @@
+import { useEnv } from '@directus/env';
+import { GraphQLSchema } from 'graphql';
+import LRUMapDefault from 'mnemonist/lru-map.js';
+import { useBus } from '../../bus/index.js';
+
+// Workaround for misaligned types in mnemonist package exports
+const LRUMap = LRUMapDefault as unknown as typeof LRUMapDefault.default;
+
+const env = useEnv();
+const bus = useBus();
+
+export const cache = new LRUMap<string, GraphQLSchema | string>(Number(env['GRAPHQL_SCHEMA_CACHE_CAPACITY'] ?? 100));
+
+bus.subscribe('schemaChanged', () => {
+	cache.clear();
+});
```

Comment:
The GQL cache should also be cleared when permissions are updated. However, permissions rely on `CACHE_AUTO_PURGE` to determine whether to purge the response cache, which is the same behaviour as other collections.

Reply:
When permissions are updated, they'll clear the system schema using the same schemaChanged message:

https://github.com/directus/directus/blob/a6d0cd5effae321a025d3fed285894e6689cbe6a/api/src/services/permissions/index.ts#L55

https://github.com/directus/directus/blob/1f4f14aaa122b4d5475b31ad76394634341f6421/api/src/cache.ts#L112

so I think this should be sufficient?

Reply:
Yeah 👍🏻  The rest behavior there is odd and will be addressed in a separate PR